### PR TITLE
Feature/android_gui_example

### PR DESCRIPTION
**Pull request by yuehua-s**
This PR introduces a complete Android GUI reinforcement learning cookbook for training vision-language models to play a number selection game using GRPO algorithm.

## Key Changes

1. **Directory Restructuring**: Renamed `examples/number_game_agent/` → `examples/android_gui_cookbook/` and moved training components to top-level `examples/` following EasyR1 conventions (prompt templates in `format_prompt/`, reward
functions in `reward_function/`).

2. **Simplified Training Script**: Created `examples/qwen2_5_vl_3b_android_gui_grpo.sh` that uses `config.yaml` as base with command-line overrides for android-gui specific parameters (kl_coef=0.04, lr=1e-5, rollout_n=8, batch_size=32).

3. **Complete Testing Framework**: Includes autonomous agent (`play_agent.py`) with ADB control (`adb_controller.py`) and unified VLM client (`vlm_client.py`) supporting both Ollama and vLLM backends for model evaluation on real Android
devices.

4. **Comprehensive Documentation**: Added unified `README.md` covering deployment (Docker/K8s), training (GRPO with Qwen2.5-VL-3B), and testing workflows, replacing scattered documentation files (ARCHITECTURE.md, PARAMETER_ALIGNMENT.md, 
etc.).

## Usage

```bash
# Deploy game
kubectl apply -f examples/android_gui_cookbook/game_docker/game.yaml

# Train model
bash examples/qwen2_5_vl_3b_android_gui_grpo.sh

# Test with agent
python examples/android_gui_cookbook/play_agent.py \
  --model-type vllm \
  --api-url http://localhost:8000 \
  --devices <android_ip>:5555

### CODE REVIEW

**Review by hiyouga — APPROVED** [review]
LGTM

**Review by hiyouga — COMMENTED** [review]
Please run `make commit && make style && make quality` to pass the style check

**Comment by yuehua-s** [comment]
> Please run `make commit && make style && make quality` to pass the style check

All style checks have passed. I've successfully run all the required checks:

Pre-commit checks (all 10 hooks passed)
  venv/bin/pre-commit run --all-files
Output: All checks passed

Code style formatting
  make style
Output: All checks passed!

Code quality checks
  make quality
Output: All checks passed!

  Changes made:
  - Removed unused variables
  - Fixed all trailing whitespace in blank lines
  - Code formatted according to ruff standards

  The code is ready for review and meets all style requirements.

**Review by hiyouga — APPROVED** [review]
